src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/BlobStorageRead.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services;
+
+import com.google.common.io.ByteStreams;
+
+import com.microsoft.azure.storage.StorageException;
+import com.microsoft.azure.storage.blob.BlobInputStream;
+import com.microsoft.azure.storage.blob.CloudBlob;
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.stream.StreamSupport;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+@Component
+public class BlobStorageRead {
```

Comment:
Can it be renamed to `DocumentProcessor` or something else ?

Reply:
extra points for javadoc comment ;)